### PR TITLE
MINOR: Cleanup storage module

### PR DESCRIPTION
**Pull request by sjhajharia**
Cleanups in Storage Module

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: storage/src/test/java/org/apache/kafka/server/log/remote/storage/RemoteTopicPartitionDirectory.java [hunk 1]
```diff
@@ -95,7 +95,7 @@ boolean delete() {
 
     void traverse(final LocalTieredStorageTraverser traverser) {
         traverser.visitTopicIdPartition(topicIdPartition);
-        listFilesets().stream().forEach(fileset -> traverser.visitSegment(fileset));
+        listFilesets().forEach(fileset -> traverser.visitSegment(fileset));
```
Comment (chia7712) [inline storage/src/test/java/org/apache/kafka/server/log/remote/storage/RemoteTopicPartitionDirectory.java]: how about: `listFilesets().forEach(traverser::visitSegment);`

File: storage/src/main/java/org/apache/kafka/storage/internals/epoch/LeaderEpochFileCache.java [hunk 2]
```diff
@@ -176,14 +176,14 @@ public Optional<EpochEntry> latestEntry() {
...
-        return entry.isPresent() ? OptionalInt.of(entry.get().epoch) : OptionalInt.empty();
+        return entry.map(epochEntry -> OptionalInt.of(epochEntry.epoch)).orElseGet(OptionalInt::empty);
     }
 
     public OptionalInt previousEpoch() {
         lock.readLock().lock();
         try {
             Optional<Map.Entry<Integer, EpochEntry>> lowerEntry = latestEntry().flatMap(entry -> Optional.ofNullable(epochs.lowerEntry(entry.epoch)));
-            return lowerEntry.isPresent() ? OptionalInt.of(lowerEntry.get().getKey()) : OptionalInt.empty();
+            return lowerEntry.map(integerEpochEntryEntry -> OptionalInt.of(integerEpochEntryEntry.getKey())).orElseGet(OptionalInt::empty);
```
Comment (chia7712) [inline storage/src/main/java/org/apache/kafka/storage/internals/epoch/LeaderEpochFileCache.java]: maybe we can eliminate local variable `lowerEntry`

```java
            return latestEntry().flatMap(entry -> Optional.ofNullable(epochs.lowerEntry(entry.epoch)))
                .map(entry -> OptionalInt.of(entry.getKey()))
                .orElseGet(OptionalInt::empty);
```

File: storage/src/main/java/org/apache/kafka/storage/internals/log/LogSegment.java [hunk 3]
```diff
@@ -459,9 +459,7 @@ public FetchDataInfo read(long startOffset, int maxSize, Optional<Long> maxPosit
 
     public OptionalLong fetchUpperBoundOffset(OffsetPosition startOffsetPosition, int fetchSize) throws IOException {
         Optional<OffsetPosition> position = offsetIndex().fetchUpperBoundOffset(startOffsetPosition, fetchSize);
-        if (position.isPresent())
-            return OptionalLong.of(position.get().offset);
-        return OptionalLong.empty();
+        return position.map(offsetPosition -> OptionalLong.of(offsetPosition.offset)).orElseGet(OptionalLong::empty);
```
Comment (chia7712) [inline storage/src/main/java/org/apache/kafka/storage/internals/log/LogSegment.java]: ditto
```java
        return offsetIndex().fetchUpperBoundOffset(startOffsetPosition, fetchSize)
            .map(offsetPosition -> OptionalLong.of(offsetPosition.offset)).orElseGet(OptionalLong::empty);
```

File: storage/src/main/java/org/apache/kafka/storage/internals/epoch/LeaderEpochFileCache.java [hunk 4]
```diff
@@ -176,14 +176,14 @@ public Optional<EpochEntry> latestEntry() {
      */
     public OptionalInt latestEpoch() {
         Optional<EpochEntry> entry = latestEntry();
-        return entry.isPresent() ? OptionalInt.of(entry.get().epoch) : OptionalInt.empty();
+        return entry.map(epochEntry -> OptionalInt.of(epochEntry.epoch)).orElseGet(OptionalInt::empty);
```
Comment (chia7712) [inline storage/src/main/java/org/apache/kafka/storage/internals/epoch/LeaderEpochFileCache.java]: ditto
```java
latestEntry().map(epochEntry -> OptionalInt.of(epochEntry.epoch)).orElseGet(OptionalInt::empty);
```

**Review by chia7712 — COMMENTED** [review]
@sjhajharia thanks for this patch. a couple of comments PTAL

**Comment by sjhajharia** [comment]
Thanks @chia7712 for the review. Addressed the comments in the new commit. Thanks!

File: storage/src/main/java/org/apache/kafka/storage/internals/log/SkimpyOffsetMap.java [hunk 5]
```diff
@@ -109,7 +109,7 @@ public long get(ByteBuffer key) throws DigestException {
         hashInto(key, hash1);
         // search for the hash of this key by repeated probing until we find the hash we are looking for or we find an empty slot
         int attempt = 0;
-        int pos = 0;
+        int pos;
```
Comment (chia7712) [inline storage/src/main/java/org/apache/kafka/storage/internals/log/SkimpyOffsetMap.java]: maybe we can move `int pos;` into the loop?

**Review by chia7712 — COMMENTED** [review]
@sjhajharia thanks for updated PR. one last comment PTAL

File: storage/src/main/java/org/apache/kafka/storage/internals/log/SkimpyOffsetMap.java (hunk 5 above)
Comment (sjhajharia) [reply storage/src/main/java/org/apache/kafka/storage/internals/log/SkimpyOffsetMap.java]: Makes sense. Moved the same in the next commit.
Thanks!

**Comment by sjhajharia** [comment]
Hey @chia7712, could you PTAL at this one final time
Thanks!

**Comment by chia7712** [comment]
@sjhajharia Could you please rebase code to trigger QA again? There are two uncompleted suits

**Comment by jlprat** [comment]
Triggering CI again as the build was aborted when running the JDK8 pipeline

**Review by chia7712 — APPROVED** [review]
LGTM

**Pull request merged**